src/sentry/flags/endpoints/logs.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime
+from typing import Any, TypedDict
+
+from rest_framework.exceptions import ParseError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.api.utils import get_date_range_from_params
+from sentry.flags.models import ActionEnum, CreatedByTypeEnum, FlagAuditLogModel
+from sentry.models.organization import Organization
+
+
+class FlagAuditLogModelSerializerResponse(TypedDict):
+    id: int
+    action: str
+    created_at: datetime
+    created_by: str
+    created_by_type: str
+    flag: str
+    tags: dict[str, Any]
+
+
+@register(FlagAuditLogModel)
+class FlagAuditLogModelSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs) -> FlagAuditLogModelSerializerResponse:
+        return {
+            "id": obj.id,
+            "action": ActionEnum.to_string(obj.action),
+            "created_at": obj.created_at.isoformat(),
+            "created_by": obj.created_by,
+            "created_by_type": CreatedByTypeEnum.to_string(obj.created_by_type),
+            "flag": obj.flag,
+            "tags": obj.tags,
+        }
+
+
+@region_silo_endpoint
+class OrganizationFlagLogIndexEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.FLAG
+    publish_status = {"GET": ApiPublishStatus.PRIVATE}
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:feature-flag-ui", organization, actor=request.user):
+            raise ResourceDoesNotExist
+
+        start, end = get_date_range_from_params(request.GET)
+        if start is None or end is None:
+            raise ParseError(detail="Invalid date range")
```

Comment:
i think we can also pass in `statsPeriod`, so `start` and `end` could be null

Reply:
e..g if `statsPeriod = 14d` then we don't need `start`/`end`

Reply:
or does `get_date_range_from_params` automatically parse `statsPeriod`?

Reply:
Yes that function parses the following headers: timeframe, timeframeStart, timeframeEnd, statsPeriod, statsPeriodStart, statsPeriodEnd, start, and end.

Reply:
I'm only interested in supporting those three headers specified in the blueprint.  Your statsPeriod widget should cover the start, end, and statsPeriod params already.  If not don't worry about supporting anything other than the statsPeriod param.  The other parameters are documented for later additions.